### PR TITLE
Avoid deprecation warning in internal use of `may_be_emitted`

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

#7580 added a deprecation warning but didn't silence it in the test suite. Just a cleanup.

### CODE REVIEW

**Comment by Pierre-Sassoulas** [comment]
Here's some context as to why we did not filter the warning: 

https://github.com/PyCQA/pylint/pull/7580#discussion_r991612938

>  we need to make a follow-up MR to make things right (give py_version instead the whole way) and I think this is a looooot of effort just for the "list-msgs-enabled", command.

We should do the refactor instead of filtering (or remove the list-msgs-enabled command ? It feels like a debug tool that no one should use if the conf is properly parsed.).

**Comment by Pierre-Sassoulas** [comment]
sys.version was what we had before, the issue is that ``list-msgs-enabled``  is wrong without the refactor (the list of message is not equal to the messages enabled if py_version != sys.version). We can remove the option entirely if fixing it is not worth it but I don't think we should knowingly keep it broken.

File: pylint/message/message_definition_store.py [hunk 1]
```diff
@@ -21,17 +21,20 @@
 class MessageDefinitionStore:
 
     """The messages store knows information about every possible message definition but
-    has no particular state during analysis.
+    has no particular state during analysis other than py_version.
```
Comment (jacobtylerwalls) [inline pylint/message/message_definition_store.py:24]: I'm not convinced of the need to add this, actually.

Comment (Pierre-Sassoulas) [reply pylint/message/message_definition_store.py:24]: Hmm, yeah the intent of the design might be broken here, maybe we do need to give the py_version all the way to list the enabled message ? On the other hand, what's the worst that could happen? Having py_version here is convenient.

Comment (jacobtylerwalls) [reply pylint/message/message_definition_store.py:24]: `py_version` is not really a state that changes during analysis, so next time I push I'll just revert this docs change.

Comment (Pierre-Sassoulas) [reply pylint/message/message_definition_store.py:24]: 👍

Comment (Pierre-Sassoulas) [reply pylint/message/message_definition_store.py:24]:
```suggestion
    has no particular state during analysis.
```

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**